rust/lance/src/dataset/take.rs

```diff
@@ -626,6 +643,55 @@ mod test {
         );
     }
 
+    #[tokio::test]
+    async fn test_take_with_deletion() {
+        let data = test_batch(0..120);
+        let write_params = WriteParams {
+            max_rows_per_file: 40,
+            max_rows_per_group: 10,
+            ..Default::default()
+        };
+        let batches = RecordBatchIterator::new([Ok(data.clone())], data.schema());
+        let mut dataset = Dataset::write(batches, "memory://", Some(write_params))
+            .await
+            .unwrap();
+
+        dataset.delete("i in (40, 77, 78, 79)").await.unwrap();
```

Comment:
Can we also test delete 0?

Reply:
I can add a test for that. What is special about deleting zero?

Reply:
just want to see if it works correctly e2e, due to we triggered a bug when `take(0)`.

Reply:
The condition you hit was there was a deletion at beginning of fragment. 40 is the beginning of the second fragment, which is why I chose that value.